transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java

```diff
@@ -113,10 +115,28 @@ public void configurePipeline(PipelineConfigurer configurer) throws IllegalArgum
       }
     }
 
+
+
     // Check if schema specified is a valid schema or no.
     try {
       Schema outputSchema = Schema.parseJson(this.config.schema);
       configurer.getStageConfigurer().setOutputSchema(outputSchema);
+
+      // When a input field is passed through to output, the type and name should be the same.
+      // If the type is not the same, then we fail.
+      if (inputSchema != null) {
+        for (Field field : inputSchema.getFields()) {
+          if (outputSchema.getField(field.getName()) != null) {
+            Schema.Type out = outputSchema.getField(field.getName()).getSchema().getType();
```

Comment:
It's better to compare the schema rather than just the schema type (e.g. the type can be enum, but they are different enums).